src/v1/account_validity.rs

```diff
@@ -0,0 +1,59 @@
+//! [GET /_synapse/admin/v1/account_validity/validity](https://github.com/matrix-org/synapse/blob/master/docs/admin_api/account_validity.rst)
+
+use std::time::SystemTime;
+
+use ruma::api::ruma_api;
+use ruma::identifiers::UserId;
+
+ruma_api! {
+    metadata: {
+        description: "Extend the account validity.",
+        method: POST,
+        name: "set_account_validity",
+        path: "/_synapse/admin/v1/account_validity/validity",
+        rate_limited: false,
+        authentication: None, // AccessToken?
+    }
+
+    request: {
+        /// user ID for the account to renew
+        pub user_id: &'a UserId,
+
+        /// This is an optional parameter, it overrides the expiration date,
+        /// which otherwise defaults to now + validity period configured at the server
+        #[serde(
+            with = "ruma::serde::time::opt_ms_since_unix_epoch",
```

Comment:
It's kind of weird but when using `serde(with)`, Option no longer defaults to `None` when absent. So there should also be `default` in this serde attribute list.